include/vctr/Expressions/BasicMath/CompareMask.h

```diff
@@ -0,0 +1,159 @@
+{
+    if constexpr (Op == CompareOp::Less)
+        return a < b;
+    if constexpr (Op == CompareOp::LessOrEqual)
+        return a <= b;
+    if constexpr (Op == CompareOp::Greater)
+        return a > b;
+    if constexpr (Op == CompareOp::GreaterOrEqual)
+        return a >= b;
+    if constexpr (Op == CompareOp::Equal)
+        return a == b;
+    if constexpr (Op == CompareOp::NotEqual)
+        return a != b;
+    return false;
+}
+
+template <size_t extent, class SrcAType, class SrcBType, CompareOp Op>
+class CompareVecVecToMask : ExpressionTemplateBase
+{
+public:
+    using value_type = bool;
+    using AValue = ValueType<SrcAType>;
+    using BValue = ValueType<SrcBType>;
+
+    static_assert (std::is_floating_point_v<AValue> && std::is_floating_point_v<BValue>,
+                   "CompareVecVecToMask currently supports floating-point element types");
+
+    VCTR_COMMON_BINARY_VEC_VEC_EXPRESSION_MEMBERS (CompareVecVecToMask, a, b)
+
+    // per-element access
+    VCTR_FORCEDINLINE constexpr value_type operator[] (size_t i) const
+    {
+        return scalarCompare<Op> (static_cast<AValue> (a[i]), static_cast<BValue> (b[i]));
+    }
+
+    //==============================================================================
+    // AVX Implementation
+    VCTR_FORCEDINLINE VCTR_TARGET ("avx") void prepareAVXEvaluation() const
+    requires has::prepareAVXEvaluation<SrcAType> && has::prepareAVXEvaluation<SrcBType>
+    {
+        a.prepareAVXEvaluation();
+        b.prepareAVXEvaluation();
+    }
+
+    //VCTR_FORCEDINLINE VCTR_TARGET ("avx") AVXRegister<value_type> getAVX (size_t i) const
+    //{
+    //    const __m256 compareResult = _mm256_cmp_ps (a.getAVX (i).value, b.getAVX (i).value, int (Op));
+    //    const int mask = _mm256_movemask_ps (compareResult);
+    //
+    //    // write bool output vector
+    //    for (unsigned i = 0; i < lanes; ++i)
+    //        dst[registerIndex + i] = ((mask >> i) & 1) != 0;
+    //}
+
+    VCTR_FORCEDINLINE VCTR_TARGET ("avx") const value_type* evalNextVectorOpInExpressionChain (value_type* dst) const
+    requires (archX64 && has::getAVX<SrcAType> && has::getAVX<SrcBType>)
+    {
+        constexpr auto lanes = size_t (8);
+        const auto n = size();
+
+        if constexpr (has::prepareAVXEvaluation<SrcAType> && has::prepareAVXEvaluation<SrcBType>)
+            prepareAVXEvaluation();
+
+        // process chunks which fit registers entirely
+        auto regCount = n / lanes;
+        for (size_t r = 0; r < regCount; ++r)
+        {
+            const auto registerIndex = r * lanes;
+
+            const auto compareResult = _mm256_cmp_ps (a.getAVX (registerIndex).value, b.getAVX (registerIndex).value, int (Op));
+            const auto mask = _mm256_movemask_ps (compareResult);
+
+            // write bool output vector from bitmask
+            for (size_t i = 0; i < lanes; ++i)
+                dst[registerIndex + i] = ((mask >> i) & 1) != 0;
```

Comment:
Here' we're doing an extra loop to pack the bitmask back into the bool vector. This seems like it breaks the entire SIMD idea, because then we could also just write a loop which compares the inputs to begin with...